### PR TITLE
Cleanup test scripts and GitHub action

### PR DESCRIPTION
**Pull request by brookslybrand**
Trying to enhance the e2e tests a bit, as well as add some extra safety around some of our scripts.

Also seeing if the GitHub action can be simplified (we'll see if I goofed something)

### CODE REVIEW

File: e2e/tsconfig.json [hunk 1]
```diff
@@ -3,6 +3,7 @@
     "baseUrl": "..", // This must be specified if "paths" is.
     "paths": {
       "@app/*": ["app/*"] // This mapping is relative to "baseUrl".
-    }
+    },
+    "noEmit": true
```
Comment (brookslybrand) [inline e2e/tsconfig.json:7]: Don't need to transpile the code, just check it

File: package.json [hunk 2]
```diff
@@ -23,9 +20,14 @@
     "test": "vitest run",
     "test:coverage": "vitest run --coverage && open ./coverage/index.html",
     "test:watch": "vitest",
+    "prep:e2e": "npm run clean && npm run build && start-server-and-test start:mocks http://localhost:3000",
+    "test:e2e": "npm run prep:e2e \"playwright test\"",
+    "test:e2e:chrome": "npm run prep:e2e \"playwright test --project=chromium\"",
+    "test:e2e:firefox": "npm run prep:e2e \"playwright test --project=firefox\"",
+    "test:e2e:webkit": "npm run prep:e2e \"playwright test --project=webkit\"",
```
Comment (brookslybrand) [inline package.json:27]: I made 2 adjustments here:
1. Prefix all e2e tests with `test:`, per our convention
2. Setup a clean, build, and start-server-and-test script with mocks to make sure that running `test:e2e` works without manually setting up the dev server.

My thought is that e2e tests should always be run against the build version anyway, definitely open to other thoughts though

File: package.json [hunk 3]
```diff
@@ -23,9 +20,14 @@
...
     "prepare": "husky install",
-    "typecheck": "tsc -b",
-    "validate": "run-p test lint typecheck test:e2e:run",
+    "typecheck": "tsc -b && tsc -b e2e",
+    "validate": "run-p test lint typecheck test:e2e",
```
Comment (brookslybrand) [inline package.json:30]: We lost checking that the e2e tests are typed correctly, and the validate script was invalid as written

File: e2e/tsconfig.json (hunk 1 above)
Comment (grantnorwood) [reply e2e/tsconfig.json:7]: Nice!

File: package.json (hunk 3 above)
Comment (grantnorwood) [reply package.json:30]: Thanks for adding that back.

**Comment by brookslybrand** [comment]
@grantnorwood and @colbywhite, would love y'all's eyes on this

I feel pretty good about the cleanup (thanks for some inspiration in #52 Colby)

Also, note that the Playwright tests are now failing. This is because of the change to run `npm run start:mocks` inside of `playwright.config.ts`. You can see [in this action](https://github.com/remix-austin/remixaustin-com/actions/runs/3658013210) everything was successful, so this will resolve once this PR is merged in

File: package.json (hunk 2 above)
Comment (brookslybrand) [reply package.json]: Thanks for the heads up, should all be working properly now.

File: package.json [hunk 4]
```diff
@@ -9,23 +9,25 @@
     "dev": "run-p dev:*",
     "dev:css": "npm run generate:css -- --watch",
     "dev:remix": "cross-env NODE_ENV=development binode --require ./mocks -- @remix-run/dev:remix dev",
-    "e2e": "npx playwright test",
-    "e2e:chrome": "npx playwright test --project=chromium",
-    "e2e:firefox": "npx playwright test --project=firefox",
-    "e2e:webkit": "npx playwright test --project=webkit",
+    "clean": "rm -rf ./build ./public/build ./node_modules/.cache ./playwright-report",
```
Comment (grantnorwood) [inline package.json:12]: 💙

**Review by grantnorwood — APPROVED** [review]
Thanks for the cleanup!  Looks much better now 🚢

**Pull request merged**